x-pack/legacy/plugins/monitoring/public/components/cluster/overview/apm_panel.js

```diff
@@ -39,48 +38,16 @@ export function ApmPanel(props) {
   const goToApm = () => props.changeUrl('apm');
   const goToInstances = () => props.changeUrl('apm/instances');
 
-  const setupModeAPMData = get(setupMode.data, 'apm');
-  let setupModeInstancesData = null;
-  if (setupMode.enabled && setupMode.data) {
-    const {
-      totalUniqueInstanceCount,
-      totalUniqueFullyMigratedCount,
-      totalUniquePartiallyMigratedCount
-    } = setupModeAPMData;
-    const hasInstances = totalUniqueInstanceCount > 0 || get(setupModeAPMData, 'detected.mightExist', false);
-    const allMonitoredByMetricbeat = totalUniqueInstanceCount > 0 &&
-      (totalUniqueFullyMigratedCount === totalUniqueInstanceCount || totalUniquePartiallyMigratedCount === totalUniqueInstanceCount);
-    const internalCollectionOn = totalUniquePartiallyMigratedCount > 0;
-    if (hasInstances && (!allMonitoredByMetricbeat || internalCollectionOn)) {
-      let tooltipText = null;
-
-      if (!allMonitoredByMetricbeat) {
-        tooltipText = i18n.translate('xpack.monitoring.cluster.overview.apmPanel.setupModeNodesTooltip.oneInternal', {
-          defaultMessage: `There's at least one server that isn't being monitored using Metricbeat. Click the flag
-          icon to visit the servers listing page and find out more information about the status of each server.`
-        });
-      }
-      else if (internalCollectionOn) {
-        tooltipText = i18n.translate('xpack.monitoring.cluster.overview.apmPanel.setupModeNodesTooltip.disableInternal', {
-          defaultMessage: `All servers are being monitored using Metricbeat but internal collection still needs to be turned
-          off. Click the flag icon to visit the servers listing page and disable internal collection.`
-        });
-      }
-
-      setupModeInstancesData = (
-        <EuiFlexItem grow={false}>
-          <EuiToolTip
-            position="top"
-            content={tooltipText}
-          >
-            <EuiLink onClick={goToInstances}>
-              <EuiIcon type="flag" color="warning"/>
-            </EuiLink>
-          </EuiToolTip>
-        </EuiFlexItem>
-      );
-    }
-  }
+  const setupModeData = get(setupMode.data, 'apm');
+  const setupModeTooltip = setupMode && setupMode.enabled
```

Comment:
Wondering why you're checking for `setupMode` here, but not in the line above. If there is a possibility of `setupMode` being undefined it might cause "accessing .data from undefined" error (in the line above)